### PR TITLE
Auth controller, AuthRoutes, applicant model password and registration

### PR DESCRIPTION
**Pull request by bartorl**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -45,6 +46,7 @@ app.use(express.static(__dirname + '/public'));
 
 //routers
 app.use('/', mainRoutes);
+app.use('/auth', authRoutes);
```
Comment (MasterCoookie) [inline app.js:49]: order routers "/" last

File: controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const Applicant = require('../models/applicantModel');
+
+const register_put = async(req, res)=>{
+    const {email, password, name, surname} = req.body;
+
+    Applicant.init().then(async() => {
+        try{
+            const applicant = await Applicant.create({email, password, name, surname});
```
Comment (MasterCoookie) [inline controllers/authController.js:8]: style: spaces

File: controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            console.log("New user %s created", email);
+
+            //req.session.newly_registered = true;
+            res.status(201).json({redirect:'login'});
```
Comment (MasterCoookie) [inline controllers/authController.js:12]: style: spaces

File: controllers/authController.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }catch(e){
+            let errors=[];
+
+            if (e.code === 11000) {
+                errors.push('Email already in use')
+            }
+    
+            if(e.errors) {
+                Object.values(e.errors).forEach(({ properties }) => {
+                    if (properties.message) {
```
Comment (MasterCoookie) [inline controllers/authController.js:22]: style: spaces

**Review by MasterCoookie — CHANGES_REQUESTED** [review]
Fix style

**Review by MasterCoookie — APPROVED** [review]

**Review by tomazaw — APPROVED** [review]

**Review by WojciechPtas — APPROVED** [review]
LGTM

**Pull request merged**